### PR TITLE
[iOS][non-icu] Clean up ICU related files from runtime pack

### PR DESCRIPTION
**Pull request by mkhamoyan**
Contributes to https://github.com/dotnet/runtime/issues/99521

### CODE REVIEW

**Review by matouskozak — APPROVED** [review]
LGTM! 
Do you know if we should do some e2e testing with Xamarin/MAUI before merging this PR?

**Comment by mkhamoyan** [comment]
> LGTM! Do you know if we should do some e2e testing with Xamarin/MAUI before merging this PR?

For xamarin-macios, I have already done the cleanup, see in [this PR](https://github.com/xamarin/xamarin-macios/pull/20830). 
If you could help with the end-to-end testing with Xamarin/MAUI, that would be greatly appreciated!

**Comment by ivanpovazan** [comment]
Shouldn't we remove the references from all places in regard to targeting Apple mobile in general?

From AppleAppBuilder:
https://github.com/dotnet/runtime/blob/2b3088b99e97f0d6d3dd24dcd41ab330757f5aa2/src/tasks/AppleAppBuilder/AppleAppBuilder.cs#L158
https://github.com/dotnet/runtime/blob/2b3088b99e97f0d6d3dd24dcd41ab330757f5aa2/src/tasks/AppleAppBuilder/Xcode.cs#L186

From: Dotnet perf pipelines:
https://github.com/dotnet/runtime/blob/2b3088b99e97f0d6d3dd24dcd41ab330757f5aa2/eng/pipelines/coreclr/templates/build-perf-sample-apps.yml#L41
https://github.com/dotnet/runtime/blob/2b3088b99e97f0d6d3dd24dcd41ab330757f5aa2/eng/pipelines/coreclr/templates/build-and-run-perf-ios-scenarios.yml#L88

etc..

**Comment by mkhamoyan** [comment]
> Shouldn't we remove the references from all places in regard to targeting Apple mobile in general?
> 
> From AppleAppBuilder:
> 
> https://github.com/dotnet/runtime/blob/2b3088b99e97f0d6d3dd24dcd41ab330757f5aa2/src/tasks/AppleAppBuilder/AppleAppBuilder.cs#L158
> 
> 
> https://github.com/dotnet/runtime/blob/2b3088b99e97f0d6d3dd24dcd41ab330757f5aa2/src/tasks/AppleAppBuilder/Xcode.cs#L186
> 
> From: Dotnet perf pipelines:
> 
> https://github.com/dotnet/runtime/blob/2b3088b99e97f0d6d3dd24dcd41ab330757f5aa2/eng/pipelines/coreclr/templates/build-perf-sample-apps.yml#L41
> 
> 
> https://github.com/dotnet/runtime/blob/2b3088b99e97f0d6d3dd24dcd41ab330757f5aa2/eng/pipelines/coreclr/templates/build-and-run-perf-ios-scenarios.yml#L88
> 
> etc..

Yes, these references should be removed across all relevant files targeting Apple mobile.
However, for this particular PR, the focus is on cleaning up ICU related files.
We can plan a separate effort to address these references in the future.

**Comment by ivanpovazan** [comment]
I agree we can split it, but if you choose to split it, then I would suggest dividing this PR into two: 
- part which directly affects our customers - product, like changes in: [src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets](https://github.com/dotnet/runtime/pull/104443/files#diff-a8c821001409a0c4d6589a59e663d4d07e53b0127b345f17640410828c580b37)
- part which cleans up our internal testing - everything else related to hybrid globalization (CI testing, performance testing, internal sample applications, builds, etc)

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets [hunk 1]
```diff
@@ -141,9 +141,6 @@ The .NET Foundation licenses this file to you under the MIT license.
       <NetCoreAppNativeLibrary Include="System.Security.Cryptography.Native.Apple" Condition="'$(_IsApplePlatform)' == 'true'" />
       <!-- Not compliant for iOS-like platforms -->
       <NetCoreAppNativeLibrary Include="System.Security.Cryptography.Native.OpenSsl" Condition="'$(StaticOpenSslLinking)' != 'true' and '$(_IsiOSLikePlatform)' != 'true'" />
-      <NetCoreAppNativeLibrary Include="icudata" Condition="'$(_IsiOSLikePlatform)' == 'true' and '$(InvariantGlobalization)' != 'true' and '$(HybridGlobalization)' != 'true'" />
-      <NetCoreAppNativeLibrary Include="icui18n" Condition="'$(_IsiOSLikePlatform)' == 'true' and '$(InvariantGlobalization)' != 'true' and '$(HybridGlobalization)' != 'true'" />
-      <NetCoreAppNativeLibrary Include="icuuc" Condition="'$(_IsiOSLikePlatform)' == 'true' and '$(InvariantGlobalization)' != 'true' and '$(HybridGlobalization)' != 'true'" />
```
Comment (matouskozak) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets:146]: We verified that these changes allow NativeAOT-iOS LibraryMode to work without setting the `<HybridGlobalization>true</HybridGlobalization>` in the csproj file.

The verification was done by manually updating the `$HOME/.nuget/packages/microsoft.dotnet.ilcompiler/9.0.0-preview.6.24327.7/build/Microsoft.NETCore.Native.Unix.targets` file and rebuilding a NativeAOT classlib project.

Comment (mkhamoyan) [reply src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets:146]: Thanks for the verification!

**Comment by mkhamoyan** [comment]
> I agree we can split it, but if you choose to split it, then I would suggest dividing this PR into two:
> 
> * part which directly affects our customers - product, like changes in: [src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets](https://github.com/dotnet/runtime/pull/104443/files#diff-a8c821001409a0c4d6589a59e663d4d07e53b0127b345f17640410828c580b37)
> * part which cleans up our internal testing - everything else related to hybrid globalization (CI testing, performance testing, internal sample applications, builds, etc)

Reverted changes in sample, testing from this PR.
Created a ticket for the second part https://github.com/dotnet/runtime/issues/105130.

File: src/tasks/AppleAppBuilder/Templates/runtime.m
Comment (ivanpovazan) [inline src/tasks/AppleAppBuilder/Templates/runtime.m:1]: Since we decided to split the PR, these changes are also not relevant :)

**Review by ivanpovazan — APPROVED** [review]
LGTM, thank you!

**Pull request merged**